src/components/TextField/TextField.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { useState } from "react";
+
+interface ITextFieldProps {
+  customClassNames?: string;
+  value?: string;
+  label?: string;
+  type: string;
+  name?: string;
+  placeholder?: string;
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+function TextField(props: ITextFieldProps) {
+  return (
+    <label>
+      {props.label}
+      <input
+        className={`${props.customClassNames}`}
+        type={props.type}
+        value={props.value}
+        placeholder={props.placeholder}
+        onChange={props.onChange}
+        name={props.name}
```

Comment:
We'll have to anticipate if `props.name` is null / undefined since its an optional property.